### PR TITLE
feat(ST-52): add file upload to firebase

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Created folder for classworks and added module, controller and service
- Updated file-upload and used firebase instead of AWS
- Downloaded packages multer and firebase 
- Updated the file-upload service layer
- Added firebase service 
- Created resources dto and updated the resource repository

## Associated Ticket Link(s)

- https://trello.com/c/Vi5GzZW1

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/resources/resources.repository.ts [hunk 1]
```diff
@@ -4,5 +4,13 @@ import { EntityRepository } from "@mikro-orm/postgresql";
 
 import { Resources } from "@/common/entities/resources.entity";
 
+import { CreateResourceDto } from "./resources.dtos";
+
 @Injectable()
-export class ResourcesRepository extends EntityRepository<Resources> {}
+export class ResourcesRepository extends EntityRepository<Resources> {
+  async createOne(resourceData: CreateResourceDto): Promise<Resources> {
```
Comment (ishmam-iqbal-sazim) [inline src/resources/resources.repository.ts:11]:
```suggestion
  async createOne(createResourceDto: CreateResourceDto): Promise<Resources> {
```

[NIT] - Probably better to keep the names consistent wherever we can

File: src/file-uploads/file-uploads.service.ts [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+  constructor(private readonly firebaseService: FirebaseService) {}
 
-  getPresignedUrl(fileUploadDto: PresignedUrlFileDto) {
-    return Promise.all(
-      fileUploadDto.files.map(async (file) => {
-        const { name, type } = file;
-        const signedUrl = await this.s3Service.getPresignedUrl(name, type);
-        return { name, type, signedUrl };
-      }),
-    );
```
Comment (ishmam-iqbal-sazim) [inline src/file-uploads/file-uploads.service.ts:17]: Please look into usage of pre signed url for file uploads

Comment (ghost) [reply src/file-uploads/file-uploads.service.ts:17]: Bhaiya, as I am using firebase as my storage, it does not go well with the pre signed url concept like S3. So I have implemented this way.

Comment (ghost) [reply src/file-uploads/file-uploads.service.ts:17]: I have another branch where I used S3 that was in the template but I was facing problem with docker and bucket, as soon as the docker is turned off I have to create the bucket again. It's not persisting, so I  have changed the storage system to firebase.

Comment (ishmam-iqbal-sazim) [reply src/file-uploads/file-uploads.service.ts:17]: There must be a way to do this for firebase as well. 
Please discuss with Ragib Bhai in the standup whether you should do that here or delegate that to another ticket.

Also, please mention the switch to firebase for storage.

**Pull request closed without merging**